apis/crd/v1alpha1/cloudentityselector_webhook.go

```diff
@@ -218,27 +218,121 @@ func (r *CloudEntitySelector) validateMatchSections() error {
+// Block same VPC ID configuration in two VMSelectors with only vpcMatch section.
+// Block same VM ID configuration in any two VMSelectors, same is not applicable for VM Name as VM Name need not be unique.
+// Block same combination of VPC ID and VM ID configuration in any two VMSelectors.
+// Block same combination of VPC ID and VM Name configuration in any two VMSelectors.
+func (r *CloudEntitySelector) validateVPCMatch(selector VirtualMachineSelector, vmIDOnlyMatch map[string]struct{},
+	vpcIDOnlyMatch map[string]struct{}, vmIDWithVpcMatch map[string]struct{},
+	vmNameWithVpcMatch map[string]struct{}) error {
+	if selector.VpcMatch.MatchID != "" {
+		if len(selector.VMMatch) == 0 {
+			// Make sure there is no VMSelector with same VPC ID with only vpcMatch section.
+			if _, found := vpcIDOnlyMatch[selector.VpcMatch.MatchID]; found {
+				return fmt.Errorf("same vpcMatch matchID %v configured in two match selectors",
+					selector.VpcMatch.MatchID)
+			}
+			vpcIDOnlyMatch[selector.VpcMatch.MatchID] = struct{}{}
+		} else {
+			for _, n := range selector.VMMatch {
+				if n.MatchID != "" {
+					// VM ID is unique across account, irrespective of VPC match same VM ID config in two VMSelector is not allowed.
+					if _, found := vmIDOnlyMatch[n.MatchID]; found {
+						return fmt.Errorf("same vmMatch matchID %v configured in two match selectors", n.MatchID)
+					}
+
+					index := n.MatchID + "/" + selector.VpcMatch.MatchID
+					if _, found := vmIDWithVpcMatch[index]; found {
+						return fmt.Errorf("same vpcMatch matchID %v and vmMatch matchID %v configured in two "+
+							"match selectors", selector.VpcMatch.MatchID, n.MatchID)
+					}
+					vmIDWithVpcMatch[index] = struct{}{}
+				} else { // Applicable for matchName.
+					index := n.MatchName + "/" + selector.VpcMatch.MatchID
+					if _, found := vmNameWithVpcMatch[index]; found {
+						return fmt.Errorf("same vpcMatch matchID %v and vmMatch matchName %v configured in two "+
+							"match selectors", selector.VpcMatch.MatchID, n.MatchName)
+					}
+					vmNameWithVpcMatch[index] = struct{}{}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// validateVMMatch function validates selector with Only vmMatch to find if the selector conflicts with existing vmSelectors.
+// Block same VM ID configuration in any two VMSelectors
+// Block same VM Name configuration in any two VMSelectors with only VMMatch section, when used along with VPCMatch it is allowed.
+func (r *CloudEntitySelector) validateVMMatch(selector VirtualMachineSelector, vmIDOnlyMatch map[string]struct{},
+	vmNameOnlyMatch map[string]struct{}, vmIDWithVpcMatch map[string]struct{}) error {
+	for _, n := range selector.VMMatch {
+		if n.MatchID != "" {
+			// VM ID is unique across account, irrespective of VPC match same VM ID configuration in two VMSelectors is not allowed.
+			if _, found := vmIDWithVpcMatch[n.MatchID]; found {
+				return fmt.Errorf("same vmMatch matchID %v configured in two match selectors", n.MatchID)
```

Comment:
Can you share the error message thrown when this condition is met?
I mean the error message shown when you do kubectl apply?

Reply:
Error from server (same vmMatch matchID i-016a83447e78d2b72 configured in two match selectors): error when creating "cloudentityselector.yaml": admission webhook "vcloudentityselector.kb.io" denied the request: same vmMatch matchID i-016a83447e78d2b72 configured in two match selectors